lib/inc/internal/facts/linux/virtualization_resolver.hpp

```diff
@@ -7,20 +7,36 @@
 #include "../resolvers/virtualization_resolver.hpp"
 #include <string>
 
+using namespace std;
+
 namespace facter { namespace facts { namespace linux {
 
     /**
      * Responsible for resolving virtualization facts.
      */
     struct virtualization_resolver : resolvers::virtualization_resolver
     {
-     protected:
+     public:
```

Comment:
Or, based on what you're doing below, use a class to inherit this and make protected members public, something like

```
struct peek_resolver : virtualization_resolver
{
    using virtualization_resolver::get_azure;
};
```